### ISSUE TITLE
System.Net.Sockets 4.1.0-rc2-23517 can not be restored

### ISSUE THREAD

**Issue by JohnACarruthers** [SapientGuardian/mysql-dnxcore50#1]
Can't use the project currently because the dep on sockets can not be restored by nuget.

Possibly missing a nuget.config in the project root directory?

**Comment by SapientGuardian** [comment]
It was premature of me to upgrade this to rc2 when I did, and that package version simply doesn't exist anymore. What are you developing with, rc1-update1 (the latest official stable release)? Or are you using the dotnet cli and actually using recent rc2 packages?

**Comment by JohnACarruthers** [comment]
I stick with the "stable" releases so still on rc1 using the dnu/dnx tools.

I tired to downgrade the socket lib ver but get NREs when connecting. After downloading your code and trying to build it against the latest beta lib on NuGet I noticed that the NetworkStream base class is missing (which probably causes the NREs).

Really unsure how you got it to work at all at this point unless the rc2 System.Net.Sockets includes NetworkStream?

**Comment by SapientGuardian** [comment]
"System.Net.Sockets": "4.1.0-beta-23516" (latest on public NuGet) has NetworkStream too. Here's a branch that builds without any rc2 dpendencies: https://github.com/SapientGuardian/mysql-dnxcore50/tree/rc1-update1
I've also pushed a myget packge for this branch if that's easier for you to test with: https://www.myget.org/F/sapientguardian/api/v2/package/mysql-dnxcore50/1.2.0-alpha

I haven't tested this at all, but it does compile :-). Let me know if it works for you.

**Comment by JohnACarruthers** [comment]
That's odd, I tried the 4.1.0-beta-23516 package from nuget and it refuses to resolve NetworkStream: ![Screenshot of Visual Studio](http://i.imgur.com/M0pFsHJ.png)

Anyway, I'll pull down that branch nuget package and give it a whirl tonight.

**Comment by JohnACarruthers** [comment]
I had to alter your nuget package to target "dotnet5.4".

The package included 2 DLLs which seemed identical (they both defined the Mysql types), so after removing one and reinstalling the package there's one remaining problem.

Attempting to use the mysql library results in a FileNotFound exception for mysql-dnxcore50.dll or one of it's dependencies.

Still no go :/

**Comment by SapientGuardian** [comment]
I'll put together an rc1-update1 project and see if I can repro this. The project I have that uses this library is already using rc2; downgrading that one isn't really practical.

**Comment by SapientGuardian** [comment]
I was able to repro the problem, and also able to get it working with a fresh rc1-update1 project. The confusion around NetworkStream is this: It seems the public nuget package for System.Net.Sockets does not have NetworkStream when you're targeting dotnet5.4, you *must* target dnxcore50 or switch to rc2. I added a commit to the rc1-update1 branch that tweaks the project.json to a state that works for the current stable bits. Using this branch, I was able to write a little web api that successfully retrieved data from a mysql database.

Hopefully rc2 will go stable soon, and this mess will be behind us.
